src/lex/replace.rs

```diff
@@ -232,8 +223,42 @@ fn replace_function(
                 }
                 break;
             }
-            // `f ;` or `f <EOF>`
-            Some(_) | None => return errors,
+            // skip any whitespaces and newlines between `f` and `(`, so that
+            // `f (` is also valid.
+            Some(Ok(Locatable {
+                data: Token::Whitespace(_),
+                ..
+            })) => {
+                let spaces = inner.next().unwrap();
+                if let Some(Ok(Locatable {
+                    data: Token::LeftParen,
+                    ..
+                })) = inner.peek()
+                {
+                    inner.next();
```

Comment:
All of these calls to `inner` are incorrect. They should first look at `incoming` and only then fall back to `inner`. The different is that `incoming` takes into account tokens from macro replacements. As a result, your change handles this incorrectly:
```c
#define f(a) 1
#define h f (2)
h
```
The expected replacement is `1`, you output `f\n (2)`.